configure.ac

```diff
@@ -2008,8 +2008,9 @@ then
 AC_MSG_CHECKING(whether $CC accepts -pthread)
 AC_CACHE_VAL(ac_cv_pthread,
 [ac_save_cc="$CC"
-CC="$CC -pthread"
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
+CC="$CC"
+CFLAGS="-pthread"
```

Comment:
if you're going to change it, you need to save and restore `CFLAGS` as is done for `CC`